src/Collection/test.js

```diff
@@ -184,6 +184,9 @@ describe('watermelondb/Collection', () => {
     const m1 = new MockTask(collection, { id: 'm1' })
```

Comment:
this test should be deleted altogether, since Collection doesn't have any real update logic

1. `Database/test` should check if `database.batch(record.prepareUpdate))` does all the things it should (compare this test with the existing test and add the missing checks if necessary)
2. `Model/test` should have a test (if it doesn't already) that when you do `record.update(x)`, both `record.prepareUpdate` was called (with `x` as arg) and `database.batch` (with `record` as arg)

the idea is to have tests cover all paths but without multiple tests testing the same thing in different places